src/components/Tour/Tour.jsx

```diff
@@ -0,0 +1,111 @@
+import './Tour.less';
+import * as d3 from 'd3'
+import React, { useMemo, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux'
+import { selectQueryTab } from '../../reducers/query/selectors';
+import { selectViewIsTourEnabled } from '../../reducers/view/selectors';
+import { Steps } from 'intro.js-react';
+import { TOUR_STEPS } from './constants/tourStepsConstants';
+import { TourButton } from './TourButton';
+import { tourToggled } from '../../actions/trends';
+
+export const Tour = () => {
+  const dispatch = useDispatch();
+  const isTourEnabled = useSelector( selectViewIsTourEnabled );
+  const tab = useSelector( selectQueryTab );
+  const steps = useMemo( () =>
+     TOUR_STEPS[tab]
+  , [ TOUR_STEPS, tab ] );
+  const stepRef = useRef();
+
+    // INTRODUCTION / TUTORIAL OPTIONS:
+  const options = {
+    disableInteraction: true,
+    scrollToElement: true,
+    scrollTo: 'tooltip',
+    showStepNumbers: false,
+    exitOnOverlayClick: false,
+    exitOnEsc: true,
+    nextLabel: 'Next',
+    prevLabel: 'Previous',
+    doneLabel: 'End Tour',
+    steps: steps
+  };
+
+
+  /**
+   * Before Change handler
+   *
+   * @param {object} ref - React component reference.
+   */
+  function handleBeforeChange( ref ) {
+    const expand = ref.current.introJs.currentStep() + 1;
+
+    // hacks to setup the page. we should figure out if we should
+    // have this in redux instead
+    if ( expand === 9 ) {
+      // search tips open/closed state should be moved to redux
+      // and when the tour start we should "reset" the page
+      document.querySelector( '.advanced-container button' ).click()
+    }
+
+    if ( expand === 16 ) {
+      // this is a hacky solution for the Trends Row chart.
+      // we need to figure out how many expandables there are
+      // figure out how many ticks there are.
+      // if not equal, it means one is open so we don't need to expand a tick
+      // i would prefer to collapse all of the bars and expand only the first.
+      // this way you can actually highlight the "View more about XYZ Product"
+      const expandables = document.querySelectorAll( '#row-chart-product .y-axis-group .tick.expandable' );
+      const ticks = document.querySelectorAll( '#row-chart-product .y-axis-group .tick' );
```

Comment:
You could eliminate the multiple row-chart-product ID lookups:

```js

const rowChartProduct = document.querySelector( '#row-chart-product' );
const expandables = rowChartProduct.querySelectorAll( '.y-axis-group .tick.expandable' );
const ticks = rowChartProduct.querySelectorAll( '.y-axis-group .tick' );

```

Reply:
Also, if there's only one `y-axis-group`, you could include that in the first query.

Reply:
i moved this logic into redux so the page state gets set to a good happy path.  We don't need to do this check anymore.